src/tabs/tokens.jsx

```diff
@@ -3,97 +3,102 @@
-      (async () => {
-        if (loginFrom === 'metamask') {
-          const res = await fetch(`https://api.opensea.io/api/v1/assets?owner=${address}&limit=${50}`, {
-            headers: {
-              'X-API-KEY': '6a7ceb45f3c44c84be65779ad2907046',
-            },
-          });
-
-          const j = await res.json();
-          const {assets} = j;
-          setNfts(assets);
-        } else if (loginFrom === 'discord') {
-          let res = await fetch(`${tokensHost}/${address}`);
-          res = await res.json();
-          res = res.map(_nft => {
-            /** Modify the response recieved from the API-Backend to match standardise format */
-            _nft.image_preview_url = _nft.hash;
-            return _nft;
-          });
-          setNfts(res);
-        }
-      })();
+      if (loginFrom === 'metamask') {
+        console.log('fetching tokens');
+        (async () => {
+          const {chainName} = await new MetamaskWallet().getChainInfo();
+          if (chainName && chainName !== 'unknown') {
+            const res = await fetch(
+              `https://nft.webaverse.com/nft?chainName=${chainName}&owner=${address}`,
+            );
+            const nfts = await res.json();
+            setNfts(nfts);
+          } else {
+            console.log('Network not supported: ', chainName);
+          }
+        })();
+      } else if (loginFrom === 'discord') {
+        (async () => {
+          const res = await fetch(
+            `https://nft.webaverse.com/nft?chainName=sidechain&owner=${address}`,
+          );
+          const nfts = await res.json();
+          setNfts(nfts);
+        })();
+      }
     }
-  }, [address, nfts, loginFrom]);
+  }, [address, loginFrom]);
 
   useEffect(() => {
     if (nfts) {
-      for (const nft of nfts) {
-        if (!nftPreviews[nft.image_preview_url]) {
-          nftPreviews[nft.image_preview_url] = 'images/object.jpg';
-          if (loginFrom === 'metamask') {
-            fetch(nft.image_preview_url).then(response => response.blob())
-              .then(imageBlob => {
-                const imageObjectURL = URL.createObjectURL(imageBlob);
-                nftPreviews[nft.image_preview_url] = imageObjectURL;
-                setNftPreviews(nftPreviews);
-              });
-          } else if (loginFrom === 'discord') {
-            /** Will be switched after Previews-Merge */
-            // preview(nft.image_preview_url, nft.ext, 'png', 100, 100).then(res => {
-            //   const imageObjectURL = URL.createObjectURL(res.blob);
-            //   nftPreviews[nft.image_preview_url] = imageObjectURL;
-            //   setNftPreviews(nftPreviews);
-            // });
+      nfts.map(async nft => {
+        try {
+          if (!nft.metadata || !nft.metadata.image) {
+            throw new Error('No image use the default one');
           }
+          const blob = await (await fetch(nft.metadata.image.replace('ipfs://', 'https://ipfs.webaverse.com/'))).blob();
```

Comment:
Is this always going to work no matter the `ipfs` url? And we will be the gateway?

Or is this only for ipfs uploaded to our end?

Reply:
this line of code checks if the image has following format:
ipfs://Qmf6FowPMZHrXdQXUyxT2YdVntprnSBECegHombaQyggd4

If yes it replaces the ipfs:// with https://ipfs.webaverse.com/

So that browser can resolve it

Reply:
So that looks wrong. How is it supposed to know the type? You could use the standard extension/file name handling we have to address this case, but its's not being used here so there is no way it will work.

Alternatively, is there a way to get the type from the HEAD? Will app IPFS servers always echo the correct content-type even if you don't use our path extension hack (not used here)?

Reply:
This piece of code is not related to totum. Here the image is fetched for preview only inside the app. No request gets sent to totum. Because it is already image. It fetches the value from "image" attribute of metadata.

the image just gets loaded as src in img element.

```
              <img
                src={nftPreviews[nft.metadata.image] || 'images/object.jpg'}
                className={styles.preview}
              />
```

Reply:
Here is where these images appear.
![sidebar](https://user-images.githubusercontent.com/13986516/155318338-9e36bc5c-3764-408b-ae8d-d0a16fb210c0.png)